### PR TITLE
Update query hooks to support queryById

### PR DESCRIPTION
**Pull request by Eunovo**
### Description

I implemented the work discussed here #730. I have tested this with sample app.

### CODE REVIEW

File: packages/datastore/datastore/src/react/hooks/query.ts [hunk 1]
```diff
@@ -19,10 +19,10 @@ const createSubscribeToMore = (model: Model, dispatch: Dispatch<Action>) => {
     };
 };
 
-const forceDelta = async (model: Model, state: ResultState, dispatch: Dispatch<Action>,  options?: IQueryOptions) => {
+const forceDelta = async (model: Model, state: ResultState, dispatch: Dispatch<Action>, options?: IQueryOptions) => {
     if (options?.forceDelta && model.replication && !state.isDeltaForced) {
-      // TODO
-      // await model.replicator.forceDeltaQuery();
+        // TODO
+        // await model.replicator.forceDeltaQuery();
```
Comment (wtrocki) [inline packages/datastore/datastore/src/react/hooks/query.ts:25]: Ah.. I have changed that. Going to be ugly conflicts here ;D

File: packages/datastore/datastore/src/react/hooks/query.ts [hunk 2]
```diff
@@ -42,27 +42,37 @@ export const useQuery = (model: Model, filter?: Filter, options?: IQueryOptions)
...
-    const query = async (filter?: Filter) => {
+    const query = async (selector?: Filter | string) => {
         if (state.isLoading) { return; }
 
         dispatch({ type: ActionType.INITIATE_REQUEST });
         try {
             await forceDelta(model, state, dispatch, options);
-            const results = await model.query(filter);
+            let results;
+            if ((typeof selector) === "string") {
```
Comment (wtrocki) [inline packages/datastore/datastore/src/react/hooks/query.ts:71]:
```suggestion
            if (selector instanceof string) {
```

File: packages/datastore/datastore/src/react/hooks/query.ts (hunk 1 above)
Comment (Eunovo) [reply packages/datastore/datastore/src/react/hooks/query.ts:25]: I didn't change anything here, I formatted the file so the spacing changed. When you merge your PR, I'll rebase this branch

Comment (wtrocki) [reply packages/datastore/datastore/src/react/hooks/query.ts:25]: Ah.. no worries. I will resolve my conflicts.

File: packages/datastore/datastore/src/react/hooks/query.ts (hunk 2 above)
Comment (Eunovo) [reply packages/datastore/datastore/src/react/hooks/query.ts:71]: @wtrocki  https://stackoverflow.com/a/203757

**Pull request merged**